lib/Remotery.c

```diff
@@ -751,6 +751,32 @@ static void AtomicSubS32(rmtAtomicS32* value, rmtS32 sub)
     AtomicAddS32(value, -sub);
 }
 
+static rmtU32 AtomicStoreU32(rmtAtomicU32* value, rmtU32 set)
+{
+#if defined(RMT_USE_C11_ATOMICS)
+    return atomic_exchange(value, set);
+#elif defined(RMT_USE_CPP_ATOMICS)
+    return value->exchange(set);
+#elif defined(RMT_PLATFORM_WINDOWS) && !defined(__MINGW32__)
+    return (rmtU32)_InterlockedExchange((long volatile*)value, (long) set);
+#elif defined(RMT_PLATFORM_POSIX) || defined(__MINGW32__)
+    return (rmtU32)__sync_lock_test_and_set(value, set);
+#endif
+}
+
+static rmtU32 AtomicLoadU32(rmtAtomicU32* value)
+{
+#if defined(RMT_USE_C11_ATOMICS)
+    return atomic_load(value);
+#elif defined(RMT_USE_CPP_ATOMICS)
+    return value->load();
+#elif defined(RMT_PLATFORM_WINDOWS) && !defined(__MINGW32__)
+    return (rmtU32)_InterlockedExchangeAdd((long volatile*)value, (long)0);
+#elif defined(RMT_PLATFORM_POSIX) || defined(__MINGW32__)
+    return (rmtU32)__sync_fetch_and_add(value, 0);
```

Comment:
Couldn't find a "load" for these two, so I added `0` instead.

Reply:
We use `LoadAcquire` and `StoreRelease` for these operations.

Unfortunately TSAN does not work with these functions as it can't recognise manual memory barriers. But then: we have no manual memory barriers in there anyway, just compiler ones.

I am happy with this change for now, but we need to look forward to merging them and specifying memory order.